multigeiger/status_led.cpp

```diff
@@ -0,0 +1,105 @@
+// show status indication via WS2812 RGB LED
+
+#include "userdefines.h"
+
+#if STATUS_LED==1 // we have a WS2812 RGB LED!
+
+#include <NeoPixelBus.h>
+
+#include "status_led.h"
+
+#define PIXEL_COUNT 1
+#define PIXEL_PIN 25
+
+RgbColor red(255, 0, 0);
+RgbColor green(0, 255, 0);
+RgbColor blue(0, 0, 255);
+RgbColor yellow(255, 255, 0);
+RgbColor white(255, 255, 255);
+RgbColor black(0, 0, 0);
+
+NeoPixelBus<NeoRgbFeature, Neo800KbpsMethod> LEDs(PIXEL_COUNT, PIXEL_PIN);
+
+static RgbColor last_col = black;
+
+void set_LED(RgbColor col)
+{
+    if(col == last_col)
+        return;  // nothing to change
+
+    LEDs.SetPixelColor(0, col);  // only 1 LED at index 0
+    LEDs.Show();
+    last_col = col;
+}
+
+void init_LED(void) {
+    LEDs.Begin();  // all LEDs off
+    LEDs.Show();
+    last_col = black;  // consistency sw state == hw state
+}
+
+#define IDX_W 2
+#define CSL 30
+
+void indicate(float radiation, unsigned int indication)
+{
+    // radiation [uSv/h] given to set the primary color R that is shown most of the time
+    // indication: 32 bit flags to indicate additional stuff, see status_led.h.
+    //
+    // this code will generate a time sequence of LED colors:
+    // index     color
+    // ------------------------------------------------------ 
+    // 0         dark (LED init)
+    // 1         indication color 1
+    // ...       reserved for more indications
+    // IDX_W     white (LED test)
+    // ...+1     radiation color
+    // ...       radiation color
+    // CSL-1     radiation color
+
+    RgbColor col;
+    int r, g, b;
+    static int index = 0;  // index counting modulo COLOR_SEQUENCE_LENGTH
+
+    switch(index) {
+        case 0:  // show a fixed dark separator after LED init
+            init_LED();  // (re-)init the LED
+            col = black;
+            break;
+        case 1:
+            r = indication & I_CONN_ERROR;
+            g = indication & I_TEST;
+            b = indication & I_HV_ERROR;
+            col = RgbColor(r ? 255 : 0, g ? 255 : 0, b ? 255 : 0);
+            break;
+        case IDX_W:  // show a fixed white separator and LED test
+            col = white;
+            break;
+        default:  // show radiation color
+            if(radiation > 1.0) {
+                col = red;
+            } else
+            if(radiation > 0.2) {
+                col = yellow;
+            } else
+            if(radiation > 0.01) {
+                col = green;
+            } else {
+                col = black;
+            }
+    }
+
+    set_LED(col);
+
+    if(++index == CSL)
+        index = 0;
+}
+
+#else // no STATUS_LED
+
+void indicate(float radiation, unsigned int indication) {
+    // do nothing
+}
```

Comment:
Due to this dummy we do not need lots of preprocessor boiler code at each place where we call indicate().

Reply:
There could be also serial output here (if there is no LED / LED is switched off).